pkg/controllers/management/auth/prtb_handler.go

```diff
@@ -50,9 +50,6 @@ type prtbLifecycle struct {
 }
 
 func (p *prtbLifecycle) Create(obj *v3.ProjectRoleTemplateBinding) (runtime.Object, error) {
-	if obj.ServiceAccount != "" {
```

Comment:
I think that you still need this block in the controller. Every controller for the PRTBs will run on every object, so this would result in entirely new bindings for PRTBs with a service account subject, which we definitely don't want.